spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/LoadBalancerAutoConfiguration.java

```diff
@@ -65,12 +67,35 @@ public void afterSingletonsInstantiated() {
 		};
 	}
 
+	@Bean
+	public LoadBalancerRequestTransformer uriReconstructingRequestTransformer() {
+		return new LoadBalancerRequestTransformer() {
+
+			@Override
+			public HttpRequest transformRequest(HttpRequest request, ServiceInstance instance,
+					LoadBalancerClient loadBalancer) {
+				return new UriReconstructingRequestWrapper(request, instance, loadBalancer);
```

Comment:
I'm not sure if your issue is with nomenclature or structure but I'll try to address both.

I chose the term transformer to mean something that takes in an HttpRequest, and spits out a HttpRequest (that could be a different object).  In this case, this is the default behaviour of returning a request that has its URI reconstructed. 

This default behaviour is provided as a LoadBalancingRequestTransformer so if an application wanted to replace it entirely it could do so. See https://github.com/spring-cloud/spring-cloud-commons/pull/164/files#diff-be91ae702a070a885afe88254370b793R184